install/build.py

```diff
@@ -367,50 +367,39 @@ def check_nvtx(compiler, settings):
     return True
 
 
-def build_shlib(compiler, source, libraries=(),
-                include_dirs=(), library_dirs=()):
-    with _tempdir() as temp_dir:
-        fname = os.path.join(temp_dir, 'a.cpp')
-        with open(fname, 'w') as f:
-            f.write(source)
+def _build_core(temp_dir, compiler, source, libraries,
+                include_dirs, library_dirs, define_macros):
+    fname = os.path.join(temp_dir, 'a.cpp')
+    with open(fname, 'w') as f:
+        f.write(source)
+    objects = compiler.compile([fname], output_dir=temp_dir,
+                               include_dirs=include_dirs,
+                               macros=define_macros)
+    try:
+        postargs = ['/MANIFEST'] if sys.platform == 'win32' else []
+        compiler.link_shared_lib(objects,
+                                 os.path.join(temp_dir, 'a'),
+                                 libraries=libraries,
+                                 library_dirs=library_dirs,
+                                 extra_postargs=postargs,
+                                 target_lang='c++')
+    except Exception as e:
+        msg = 'Cannot build a stub file.\nOriginal error: {0}'.format(e)
+        raise Exception(msg)
 
-        objects = compiler.compile([fname], output_dir=temp_dir,
-                                   include_dirs=include_dirs)
 
-        try:
-            postargs = ['/MANIFEST'] if sys.platform == 'win32' else []
-            compiler.link_shared_lib(objects,
-                                     os.path.join(temp_dir, 'a'),
-                                     libraries=libraries,
-                                     library_dirs=library_dirs,
-                                     extra_postargs=postargs,
-                                     target_lang='c++')
-        except Exception as e:
-            msg = 'Cannot build a stub file.\nOriginal error: {0}'.format(e)
-            raise Exception(msg)
+def build_shlib(compiler, source, libraries=(),
+                include_dirs=(), library_dirs=(), define_macros=None):
+    with _tempdir() as temp_dir:
+        _build_core(temp_dir, compiler, source, libraries,
+                    include_dirs, library_dirs, define_macros)
 
 
 def build_and_run(compiler, source, libraries=(),
-                  include_dirs=(), library_dirs=()):
+                  include_dirs=(), library_dirs=(), define_macros=None):
     with _tempdir() as temp_dir:
-        fname = os.path.join(temp_dir, 'a.cpp')
-        with open(fname, 'w') as f:
-            f.write(source)
-
-        objects = compiler.compile([fname], output_dir=temp_dir,
-                                   include_dirs=include_dirs)
-
-        try:
-            postargs = ['/MANIFEST'] if sys.platform == 'win32' else []
-            compiler.link_executable(objects,
```

Comment:
`build_and_run` calls `link_executable`, but `build_shlib ` calls `link_shared_lib `. You cannot change it.

Reply:
OK, I will fix it.